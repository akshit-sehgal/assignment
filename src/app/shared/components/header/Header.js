import React from 'react';
import PropTypes from 'prop-types';
import { HEADER_LABELS } from './constants';
import './style.scss';
class Header extends React.PureComponent {
    componentDidMount() {
        const { headerImg, fetchHeaderCarouselData } = this.props;
        if (!headerImg) {
            fetchHeaderCarouselData();
        }
    }

    render() {
        const { headerImg, navigate } = this.props;
        return (
            <div className="header-container">
                <div className="header__back-holder">
                    {navigate &&
                        <button className="header__back" onClick={navigate}>
                            <i className="material-icons">
                                keyboard_backspace
                        </i>
                            <span>{HEADER_LABELS.BACK_BTN}</span>
                        </button>
                    }
                </div>
                <img className="header__branding" src={headerImg} alt='Logo' />
                <div className="header__name">{HEADER_LABELS.DESCRIPTION}</div>
            </div>
        )
    }
}
Header.propTypes = {
    headerImg: PropTypes.string.isRequired,
    navigate: PropTypes.func,
    fetchHeaderCarouselData: PropTypes.func.isRequired
}
export default Header;