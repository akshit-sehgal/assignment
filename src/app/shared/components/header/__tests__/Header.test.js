import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

describe('Testing <Header/>', () => {
    let props;
    beforeEach(() => {
        props = {
            headerImg: 'https://images.google.com/1.png',
            navigate: jest.fn(),
            fetchHeaderCarouselData: jest.fn()
        }
    })
    it('should render the default view', () => {
        const wrapper = shallow(<Header {...props} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.header__back').length).toEqual(1);
        expect(wrapper.find('.header__branding').prop('src')).toEqual(props.headerImg);
    });
    it('should call navigate when back button is clicked', () => {
        const wrapper = shallow(<Header {...props} />);
        wrapper.find('.header__back').simulate('click');
        expect(props.navigate).toHaveBeenCalledTimes(1);
    });
    it('should call fetchHeaderCarouselData tp fetch data', () => {
        props.headerImg = '';
        props.navigate = null;
        const wrapper = shallow(<Header {...props} />);
        expect(props.fetchHeaderCarouselData).toHaveBeenCalledTimes(1);
        expect(wrapper).toMatchSnapshot();
    });

});