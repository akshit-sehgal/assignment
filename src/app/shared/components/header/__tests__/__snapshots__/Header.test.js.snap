// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Header/> should call fetchHeaderCarouselData tp fetch data 1`] = `
<div
  className="header-container"
>
  <div
    className="header__back-holder"
  />
  <img
    alt="Logo"
    className="header__branding"
    src=""
  />
  <div
    className="header__name"
  >
    Welcome to tokopedia.com!
  </div>
</div>
`;

exports[`Testing <Header/> should render the default view 1`] = `
<div
  className="header-container"
>
  <div
    className="header__back-holder"
  >
    <button
      className="header__back"
      onClick={[MockFunction]}
    >
      <i
        className="material-icons"
      >
        keyboard_backspace
      </i>
      <span>
        Back
      </span>
    </button>
  </div>
  <img
    alt="Logo"
    className="header__branding"
    src="https://images.google.com/1.png"
  />
  <div
    className="header__name"
  >
    Welcome to tokopedia.com!
  </div>
</div>
`;
